packages/embark-test-runner/src/reporter.js

```diff
@@ -55,7 +55,14 @@ class EmbarkSpec extends Base {
       if (self.txDetails) {
         self.embarkEvents.request('contracts:contract', receipt.className, (contract) => {
           if (contract) {
-            self.contracts.push(contract);
+            let index = self.contracts.findIndex(c => c.className === contract.className);
+            // It's possible to deploy the same contract multiple times per test, so we need
+            // to make sure we replace the existing one with the new one.
+            if (index > -1) {
+              self.contracts = self.contracts.map(c => c.className === contract.className ? contract : c);
```

Comment:
True, not a huge win.

I just had another thought. Since you already have the array index and the logic to find it is the same logic used in the map function, could we do:

```suggestion
              self.contracts[index] = contract;
```

Reply:
Sorry, I just realized this suggestion-comment is a separate thread of conversation. I'm following up on https://github.com/embark-framework/embark/pull/1714/files#r301673121.

Reply:
That's a great suggestion! D'oh :D